plugins/refetch-interval/src/index.ts

```diff
@@ -0,0 +1,141 @@
+import type { PiniaColadaPluginContext } from '@pinia/colada'
+import { toValue } from 'vue'
+/**
+ * @module @pinia/colada-plugin-refetch-interval
+ */
+
+/**
+ * Options for the Pinia Colada Refetch Interval plugin.
+ */
+export interface RefetchIntervalOptions {
+  /**
+   * If set to a number, queries will continuously refetch at this frequency in milliseconds.
+   * If set to false, no automatic refetching will occur.
+   */
+  refetchInterval?: number | false
+  /**
+   * If true, queries will continue to refetch while their tab/window is in the background.
+   */
+  refetchIntervalInBackground?: boolean
+}
+
+interface IntervalEntry {
+  timeoutId: ReturnType<typeof setTimeout>
+  lastUpdated: number
+}
+
+const REFETCH_INTERVAL_OPTIONS_DEFAULTS = {
+  refetchInterval: false,
+  refetchIntervalInBackground: false,
+} satisfies Required<RefetchIntervalOptions>
+
+/**
+ * Plugin that adds the ability to refetch queries at regular intervals.
+ *
+ * @param globalOptions - global options for the intervals
+ */
+export function PiniaColadaRefetchInterval(
+  globalOptions?: RefetchIntervalOptions,
+): (context: PiniaColadaPluginContext) => void {
+  const defaults = { ...REFETCH_INTERVAL_OPTIONS_DEFAULTS, ...globalOptions }
+
+  return ({ queryCache }) => {
+    const intervalMap = new Map<string, IntervalEntry>()
+
+    let isDocumentVisible = typeof document !== 'undefined'
```

Comment:
I'm wondering if this makes sense at this level. It could be set in the `enabled` option so it feels a bit redundant here, the overlap between options could prove difficult in the future to debug.